src/rpc/ws_client/mod.rs

```diff
@@ -108,271 +105,17 @@ impl HandleMessage for SubscriptionHandler {
 
 		match value["id"].as_str() {
 			Some(_idstr) => {},
-			_ => {
-				// subscriptions
-				debug!("no id field found in response. must be subscription");
-				debug!("method: {:?}", value["method"].as_str());
-				match value["method"].as_str() {
-					Some("state_storage") => {
```

Comment:
No need to distinguish between subscription methods - we can simply return the result for any subscription. The differentiation is handled one level below in the `SubscriptionWrapper::next()` where a `Notification` is deserialized.